assets/js/components/button.js

```diff
@@ -82,9 +82,9 @@ class Button extends Component {
 				aria-controls={ ariaControls }
 				role={ 'a' === SemanticButton ? 'button' : undefined }
 			>
-				{ icon && icon }
+				{ icon }
 				<span className="mdc-button__label">{ children }</span>
-				{ trailingIcon && trailingIcon }
+				{ trailingIcon }
```

Comment:
Why were these changed? I'm not saying this is necessarily a problem, but as far as I remember JSX behaves slightly different from regular JS logic for those types of comparisons - there were cases like this one before which resulted in unexpected behavior. Are we 100% sure removing these checks won't render something like `undefined` or `0` or `false` out as a string in the HTML?

Reply:
@felixarntz it is not necessary because if trailingIcon is `undefined` it will be `null` (because it's a default value for trailingIcon) and will render nothing. If it is `false`, it will also render nothing too because react doesn't render an element if it's `false`. If you pass `0`, it will render `0` even with `{ trailingIcon && trailingIcon }` apporach because `0 && 0` always returns `0`.

Reply:
@felixarntz I don't think there is anything JSX specific about this behavior. See the docs for it here: https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

If I recall, I believe the problem that happened before was with settings where the default value was `'0'`. In JS, a string zero is considered a truthy value, so it will be output. E.g. 
![image](https://user-images.githubusercontent.com/1621608/91159686-0f737680-e6d1-11ea-8f82-add7be931a0f.png)

The `&&` causes the result of the expression return the first falsy value, or the last truthy value, whichever comes first. 
![image](https://user-images.githubusercontent.com/1621608/91159966-7e50cf80-e6d1-11ea-9e9b-ff32cd299483.png)


In this case, I think @eugene-manuilov is correct to simplify this logic as it is unnecessarily duplicated.